lib/checkNewCronJobs.js

```diff
@@ -0,0 +1,111 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const {
+  serverJobAdmins
+} = require('../userWhitelist.json');
+const {
+  getAllChangedFiles,
+  pingAndAssignUsers
+} = require('./utils');
+const CRON_REGISTRY_FILENAME = 'main_cron.py';
+const CRON_TEST_FILENAME = 'core/controllers/cron_test.py';
+const CRON_JOB_FILE = 'core/controllers/cron.py';
+const WIKI_LINK = 'https://github.com/oppia/oppia/wiki/' +
+ 'Testing-jobs-and-other-features-on-production';
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ */
+const isCronFileChanged = (changedFiles) => {
+  const file = (fileObject) => fileObject.filename === CRON_JOB_FILE;
+  return changedFiles.some(file);
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ */
+const hasAddedTests = (changedFiles) => {
+  const file = (fileObject) =>
+    fileObject.filename === (CRON_REGISTRY_FILENAME || CRON_TEST_FILENAME);
+  return changedFiles.some(file);
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ */
+const getRegistryReminder = (changedFiles) => {
+  let registryReminder = '';
+  if (!hasAddedTests(changedFiles)) {
+    registryReminder +=
+      ' please add the new test and URL redirects for new cron jobs';
+  }
+  return registryReminder;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ * @param {string} prAuthor - Author of the Pull Request
+ */
+const getCommentBody = (changedFiles, prAuthor) => {
+  const registryReminder = getRegistryReminder(changedFiles);
+  const newLineFeed = '<br>';
+  const serverJobsForm = 'server jobs form'.link(
+    'https://goo.gl/forms/XIj00RJ2h5L55XzU2');
+  const wikiLinkText = (
+    'this guide'.link(WIKI_LINK));
+
+  const serverAdminPing = (
+    'Hi @' + serverJobAdmins.join(', @') +
+          ', PTAL at this PR, ' + 'it adds a new cron job.'
+  );
+
+  const prAuthorPing = 'Also @' + prAuthor + registryReminder +
+          ' It seems you have added or edited a CRON job,' +
+          'if so please request a testing of this CRON job ' +
+          'with this ' + serverJobsForm +
+          ' Please refer to ' + wikiLinkText + ' for details.';
+  return (serverAdminPing + newLineFeed + prAuthorPing +
+          newLineFeed + 'Thanks!');
+};
+
+/**
+ * @param {import('probot').Context} context
+ */
+const checkForNewCronJob = async (context) => {
+  /**
+   * @type {import('probot').Octokit.PullsGetResponse} pullRequest
+   */
```

Comment:
Why is this here?

Reply:
I think it's for type inference.

Reply:
Other files also used the same pattern so I added it.It specifies the type here.